tests/targets/_internal/test_board_database.py

```diff
@@ -16,52 +18,65 @@
 class TestGetOnlineBoardData(TestCase):
     """Tests for the method `board_database.get_online_board_data`."""
 
+    @pytest.fixture(autouse=True)
+    def add_caplog(self, caplog):
+        self._caplog = caplog
```

Comment:
I'm not sure this is right. It's not necessary to make our own caplog fixture, is it?

See how we use caplog at https://github.com/ARMmbed/mbed-tools/blob/master/tests/build/_internal/config/test_config.py#L114-L119 for instance. There is no `add_caplog()` or `self._caplog` there, but we are able to use the pre-made pytest built-in `caplog`.

Reply:
It's because of it being used with `unittest.TestCase` methods, which can't directly receive the pytest fixture as an argument. I could refactor it so that they're not `TestCase` methods, which would remove the need for `self._caplog`.

Reply:
We've been moving away from unittest in most of the other modules. I'd recommend we refactor to just use pytest directly as an early commit in this PR.